enhancements/console/dynamic-plugins.md

```diff
@@ -0,0 +1,369 @@
+        "flag": "EXAMPLE",
+        "model": {
+          "group": "example.com",
+          "version": "v1",
+          "kind": "ExampleModel"
+        }
+      }
+    }
+  ]
+}
+```
+
+#### Plugin Manifest Properties
+
+| Property | Description |
+| ----- | -------- |
+| `name: string` | Used as a unique identifier. Each plugin must have a unique name. |
+| `version: string` | The version of the plugin. Version must be parsable by node-semver, eg. “1.2.3” |
+| `pluginAPI: string` | Semver range of compatible OpenShift console versions. |
+| `displayName?: string` | User friendly display name. |
+| `description?: string` | The description of the plugin. |
+| `extensions: Extension[]` | List of extensions contributed by the plugin. |
+| `dependencies?: { [pluginName: string]: string }` | A plugin which depends on other plugins must explicitly define their dependencies. |
+
+(Types are expressed as TypeScript types.)
+
+##### Extension<P = JSON>
+
+| Property | Description |
+| ----- | -------- |
+| `id: string` | Unique identifier of this extension. |
+| `type: string` | The `type` identifies this extension of a particular extension type. |
+| `properties: P` | The extension properties JSON. |
+| `flags?: Partial<{ required: string[]; disallowed: string[]; }>` | Reference to the flags used to trigger the enablement of this extension. |
+
+An extension may contain code references, encoded as object literal `{
+$codeRef: string }`. The value of `$codeRef` is `moduleName.exportName` or
+`moduleName` (equivalent to `moduleName.default`). Console will resolve the
+code references when needed.
+
+### Lazy Loading
+
+Console uses
+[code splitting](https://webpack.js.org/guides/code-splitting/) to only load
+JavaScript code when needed. Likewise, dynamic plugin code will only load when
+needed. For instance, if the plugin contributes a list page, the plugin code
+for that list page only loads when the user visits the page. This improves
+performance and guards against bugs in plugins the user is not actively using.
+
+Console performs feature detection on initial load. This is used today to
+enable or disable parts of the UI based on features in the cluster and is
+typically driven by the resources found during API discovery. Dynamic plugins
+will also be able to declare console feature flags and will only load if the
+flag is enabled. For instance, the KubeVirt plugin will only load if the
+`VirtualMachine` resource is present.
+
+### Error Handling
+
+Console will guard against runtime errors in plugins. All plugin components
+will be surrounded by [React error boundaries](https://reactjs.org/docs/error-boundaries.html).
+This prevents an uncaught error from causing the application to white-screen
+and break. If a plugin service is unavailable, console will not load the plugin
+and log an error.
+
+Additionally, the console can contain a `?disable-plugins` query parameter.
+When this parameter is present, no dynamic plugins are loaded.
```

Comment:
you don't want to make this take the names of the plugins to disable?  (with an optional `all` arg)?  That seems like it could be useful for debugging/support (disabling specific plugins instead of all or none of them)

Reply:
@spadgett you thumbs-upped this but i don't think you addressed it in your updates.

Reply:
Sorry, missed that. I'll update the PR.